hs_access_control/models/group.py

```diff
@@ -162,44 +300,35 @@ def get_resources_with_explicit_access(self, this_privilege):
-                                                .filter(r2grp__group=self.group,
-                                                        r2grp__privilege__lt=this_privilege))
+            return BaseResource.objects.none()  # groups cannot own resources
 
         elif this_privilege == PrivilegeCodes.CHANGE:
             # CHANGE does not include immutable resources
             return BaseResource.objects.filter(raccess__immutable=False,
                                                r2grp__privilege=this_privilege,
-                                               r2grp__group=self.group)\
-                                       .exclude(pk__in=BaseResource.objects
-                                                .filter(r2grp__group=self.group,
-                                                        r2grp__privilege__lt=this_privilege))
+                                               r2grp__group=self.group)
+            # there are no excluded resources; maximum privilege is CHANGE
 
-        else:  # this_privilege == PrivilegeCodes.ViEW
+        else:  # this_privilege == PrivilegeCodes.VIEW
             # VIEW includes CHANGE & immutable as well as explicit VIEW
-            view = BaseResource.objects.filter(r2grp__privilege=this_privilege,
-                                               r2grp__group=self.group)\
-                .exclude(pk__in=BaseResource.objects.filter(r2grp__group=self.group,
-                                                            r2grp__privilege__lt=this_privilege))
-
-            immutable = BaseResource.objects.filter(raccess__immutable=True,
-                                                    r2grp__privilege=PrivilegeCodes.CHANGE,
-                                                    r2grp__group=self.group)\
-                                            .exclude(pk__in=BaseResource.objects.filter(
-                                                raccess__immutable=True,
-                                                r2grp__group=self.group,
-                                                r2grp__privilege__lt=PrivilegeCodes.CHANGE))
-
-            return BaseResource.objects.filter(Q(pk__in=view) | Q(pk__in=immutable)).distinct()
+            excluded = BaseResource.objects.filter(raccess__immutable=False,
+                                                   r2grp__privilege=PrivilegeCodes.CHANGE,
+                                                   r2grp__group=self.group).values('pk')
+            return BaseResource.objects.filter(Q(r2grp__privilege=PrivilegeCodes.VIEW,
+                                                 r2grp__group=self.group) |
+                                               Q(raccess__immutable=True,
+                                                 r2grp__privilege=PrivilegeCodes.CHANGE,
+                                                 r2grp__group=self.group))\
+                                       .exclude(pk__in=Subquery(excluded)).distinct()
```

Comment:
It seems this `Subquery` can return multiple values ('pk'). In that case it will throw exception from DB. Ref: https://docs.djangoproject.com/en/1.11/ref/models/expressions/#limiting-a-subquery-to-a-single-column

Reply:
Igone my above comment.